ui/src/authors/containers/PublicationSelectContainer.jsx

```diff
@@ -5,14 +5,20 @@ import {
   setPublicationSelection,
   setPublicationsClaimedSelection,
   setPublicationsUnclaimedSelection,
+  setPublicationsCanNotClaimSelection,
 } from '../../actions/authors';
 
 const stateToProps = (state, { recordId }) => ({
   checked: state.authors.get('publicationSelection').has(recordId),
 });
 
-const dispatchToProps = (dispatch, { recordId, claimed }) => ({
+const dispatchToProps = (dispatch, { recordId, claimed, canClaim }) => ({
   onChange(event) {
+    if (!canClaim) {
```

Comment:
I would split it into 3 (or maybe 4) different dispatches

Reply:
again, we need to have somewhere a logic which actions we want to do `onChange`, I don't know which place would be better